Make scrollbar move with tts

Burge337 opened the pull request:
This fixes issues #36 and #37.

Note: a bug appeared somewhere along the line that breaks the TTS when clicking on word tiles in the sentence bar and in the word holder. This is fixed in the resubmit-sentence branch, so don't worry about it here. It should be fixed when that branch is merged into master.

Burge337 committed:
Implement scrollbar movement to match TTS and highlighting

Also:
Made the word tiles in the sentence bar slightly less wide.
Made the sentence bar slightly taller than the words it holds.
Made the sentence bar's slider properly shrink when dragging words out of the sentence bar.
Made the "reveal" animation at the bottom wait a bit before revealing to leave time for the pipes to animate in the future.
Made both the TTS button and submit sentence button shift the scrollbar left when they start.
Important note:
The functionality of the matching scrollbar movement to TTS is based on the assumption that exactly 6 tiles will fill up all of the sentence bar. If we end up changing around the sentence bar size or the sizes of the word tiles it houses, we'll have to make sure those changes are reflected in TextToSpeechHandler.cs

Burge337 committed:
Adjust the number of tiles read before scrolling the sentence bar

Previously the sentence bar didn't begin scrolling until TTS reached the last word in view, where it would then reveal only the next word. This commit changes things so that we instead begin scrolling (if scrolling is possible) after reading 4 word tiles, (of the 6 in view) so that kids are afforded the context of the nearby words.

fixes #36 #37

kklamberty reviewed (changes requested):
We discovered that the animation for saving a sentence that needs to scroll sometimes gets a little choppy. We are concerned it may be a performance issue with the animation itself. @Burge337 is going to explore if there is a difference with Lean Tween (or however that is spelled) compared to any hand made animation approach that might be in use there. This looks like it allows for scrolling all the way to the edges, though, which fixes some of the problem that this PR aims to fix.

Burge337 committed:
Change implementation of revealSentenceAnimation to hopefully be faster

I've stored the approximate time it takes for tts to speak through everything in a variable so it only needs to be calculated once. I'm wondering if maybe the weird stuttering lag was from having passed tts.getApproxSpeechTime to the lean tween moveX function. I could see it maybe having to invoke that method a bunch of times when any part of the moveX function that wanted to know how long the duration of the animation is supposed to be would have to run that method each time.

Burge337 committed:
Create LeanTween implemenation of ClearTiles method

I'm hoping that this will fix the performance issues that have cropped up  with submit sentence stuttering.

Burge337 commented:
Having looked into the stuttering issue with @kklamberty, it seems like there may be some funny stuff happening with the garbage collection behind the scenes. My hunch is that the cleartiles method is the source of the lag, where it deletes tiles and resizes the list every time a tile animates. The picture attached shows these spikes that coincide with the TTS speaking.
![Profiler](https://user-images.githubusercontent.com/57874855/108641076-f7ec2f00-7462-11eb-8824-a0cfaaddb5a1.PNG)

kklamberty reviewed (approved):
The good news is that the stuttering in the animation does seem to be fixed. Yay!
